tests/Integration/Core/Module/ModuleRepositoryTest.php

```diff
@@ -55,11 +56,46 @@ protected function setUp(): void
         $translator = $this->createMock(Translator::class);
         $translator->method('trans')->willReturnArgument(0);
 
+        $hookManager = $this->createMock(HookManager::class);
+        $hookExecMethodMock = function () {
+            // cf. HookManager::exec() method signature
+            list(
+                $hook_name,
+                $hook_args,
+                $id_module,
+                $array_return,
+                $check_exceptions,
+                $use_push,
+                $id_shop
+                ) = func_get_args();
+
+            // This mock represents a module that :
+            // - overrides `dummy_payment` module `fullDescription` attributes
+            // - adds `testAttribute` attributes to `dummy_payment` module
+            // when 'actionListModules' hook called
+            if ($hook_name == 'actionListModules') {
+                return [
+                    'ps_distributionapiclient' => [
+                        [
+                            'name' => 'dummy_payment',
+                            'fullDescription' => 'overridden full description',
+                            'testAttribute' => 'added value',
+                        ],
+                    ],
+                ];
+            } else {
+                return [];
+            }
+        };
+        $hookManager->method('exec')->willReturn(
+            $this->returnCallback($hookExecMethodMock)
```

Comment:
```suggestion
            $this->returnCallback(function(....))
```
Also I'm not sure it is useful to store the callback in a variable since it's never used anywhere else, you can create the callback directly here

Reply:
In this case, it is just a matter of taste. For reading/debugging/tests I am more comfortable to separate things when there are too much inclusions.

If it is important to you, I can change. Let me know.